cmake/traccc-compiler-options-cuda.cmake

```diff
@@ -24,7 +24,7 @@ if( "${CMAKE_CUDA_COMPILER_ID}" MATCHES "NVIDIA" )
 endif()
 
 # Set the CUDA architecture to build code for.
-set( CMAKE_CUDA_ARCHITECTURES "52" CACHE STRING
+set( CMAKE_CUDA_ARCHITECTURES "75" CACHE STRING
```

Comment:
Yesh. Unfortunately the include order in the project makes this indeed ineffective. 😦

In a separate PR we should change the setup a little. Moving such global settings simply to the main `CMakeLists.txt` file. Such that they would be guaranteed to precede whatever version some of the externals are setting up.

Note that I've been setting `CMAKE_CUDA_ARCHITECTURES` in `CMakeUserPresets.json` recently to set for instance `"native"` in my local builds. (Just to give ideas.)

Reply:
I'd say the superior option would really be to have the dependent projects not set any global CMake variables!